### ISSUE TITLE
dbExistsTable(conn, name) is not TRUE  in ORACLE connection

### ISSUE THREAD

**Issue by capitantyler** [r-dbi/odbc#702]
I am trying to insert rows in existent table, but dbExistsTable(conn, name) returns FALSE.

These are my tables (I am consulting for "TIPO_CAMBIO"
```r
DBI::dbListTables(conn)
#[1] "COMPRAS_MD"             "TRX_MENSUAL_APROBADAS" 
#[3] "TIPO_CAMBIO2"           "TIPO_CAMBIO"           
#[5] "TIPO_CAMBIO3"           "TRX_MENSUAL_REVERSADAS"
#[7] "MTCARS"                 "mtcars"          

# check existence VIA SQL
res <- DBI::dbGetQuery(
  conn,
  "Select 1 from  sys.all_tables where owner = 'USR_REPORTE_FRAUDES' and table_name = 'TIPO_CAMBIO'" 
)
res
#  1
#1 1
```
But dbExistsTable(conn, "TIPO_CAMBIO") is FALSE

```r
 dbExistsTable(conn, "TIPO_CAMBIO")
# [1] FALSE
dbExistsTable(conn, Id(
    schema = "USR_REPORTE_FRAUDES", table = "TIPO_CAMBIO"
))
#[1] FALSE
 dbExistsTable(conn, Id(
    owner = "USR_REPORTE_FRAUDES", table = "TIPO_CAMBIO"
))
#[1] FALSE
```

These are my connection string and my TNSNAMES which I has been connecting succesfully:

PRULES =
  (DESCRIPTION =
    (ADDRESS = (PROTOCOL = TCP)(HOST = exac2-scan1.lorenipsum.com.ar)(PORT = MyPort))
    (CONNECT_DATA =
      (SERVER = DEDICATED)
      (SERVICE_NAME = myServiceName)
    )
  )

Driver={{Oracle en OraClient12Home1}};DBQ=exac2-scan1.lorenipsum.com.ar:MyPort/myServiceName;UID={usr};PWD={pwd};

<details>
more info
```r
 DBI::dbGetInfo(conn)
# $dbname
# [1] ""
# 
# $dbms.name
# [1] "Oracle"
# 
# $db.version
# [1] "19.00.0000"
# 
# $username
# [1] "USR_REPORTE_FRAUDES"
# 
# $host
# [1] ""
# 
# $port
# [1] ""
# 
# $sourcename
# [1] "Default"
# 
# $servername
# [1] "exac2-scan1.lorenipsum.com.ar:1521/myServiceName"
# 
# $drivername
# [1] "SQORA32.DLL"
# 
# $odbc.version
# [1] "03.80.0000"
# 
# $driver.version
# [1] "12.01.0020"
# 
# $odbcdriver.version
# [1] "03.52"
# 
# $supports.transactions
# [1] TRUE
# 
# $getdata.extensions.any_column
# [1] TRUE
# 
# $getdata.extensions.any_order
# [1] TRUE
# 
# attr(,"class")
# [1] "Oracle"      "driver_info" "list"   
```

```r
devtools::session_info()
# ─ Session info ───────────────────────────────────────────────────────────
# setting  value
# version  R version 4.3.2 (2023-10-31 ucrt)
# os       Windows 10 x64 (build 19045)
# system   x86_64, mingw32
# ui       RStudio
# language (EN)
# collate  Spanish_Argentina.utf8
# ctype    Spanish_Argentina.utf8
# tz       America/Buenos_Aires
# date     2023-12-20
# rstudio  2023.12.0+369 Ocean Storm (desktop)
# pandoc   3.1.11 @ C:\\Users\\EC_E91~1\\AppData\\Local\\r-pandoc\\r-pandoc\\31F387~1.11\\pandoc.exe
# 
# ─ Packages ───────────────────────────────────────────────────────────────
# package     * version     date (UTC) lib source
# bit           4.0.5       2022-11-15 [1] CRAN (R 4.2.1)
# bit64         4.0.5       2020-08-30 [1] CRAN (R 4.2.1)
# blob          1.2.4       2023-03-17 [1] CRAN (R 4.2.3)
# box           1.1.3       2023-05-02 [1] CRAN (R 4.3.0)
# cachem        1.0.8       2023-05-01 [1] CRAN (R 4.3.0)
# cli           3.6.2       2023-12-11 [1] CRAN (R 4.3.2)
# colorspace    2.1-0       2023-01-23 [1] CRAN (R 4.2.2)
# crayon        1.5.2       2022-09-29 [1] CRAN (R 4.2.1)
# data.table  * 1.14.10     2023-12-08 [1] CRAN (R 4.3.2)
# DBI         * 1.1.99.9000 2023-12-20 [1] Github (r-dbi/DBI@397bb7c)
# dbplyr      * 2.4.0       2023-10-26 [1] CRAN (R 4.3.2)
# devtools      2.4.5       2022-10-11 [1] CRAN (R 4.2.1)
# digest        0.6.33      2023-07-07 [1] CRAN (R 4.3.1)
# dplyr       * 1.1.4       2023-11-17 [1] CRAN (R 4.3.2)
# ellipsis      0.3.2       2021-04-29 [1] CRAN (R 4.2.1)
# fansi         1.0.6       2023-12-08 [1] CRAN (R 4.3.2)
# fastmap       1.1.1       2023-02-24 [1] CRAN (R 4.2.2)
# forcats     * 1.0.0       2023-01-29 [1] CRAN (R 4.2.2)
# fs            1.6.3       2023-07-20 [1] CRAN (R 4.3.1)
# generics      0.1.3       2022-07-05 [1] CRAN (R 4.2.1)
# ggplot2     * 3.4.4       2023-10-12 [1] CRAN (R 4.3.1)
# glue        * 1.6.2       2022-02-24 [1] CRAN (R 4.2.1)
# gtable        0.3.4       2023-08-21 [1] CRAN (R 4.3.1)
# hms           1.1.3       2023-03-21 [1] CRAN (R 4.2.3)
# htmltools     0.5.7       2023-11-03 [1] CRAN (R 4.3.2)
# htmlwidgets   1.6.4       2023-12-06 [1] CRAN (R 4.3.2)
# httpuv        1.6.13      2023-12-06 [1] CRAN (R 4.3.2)
# later         1.3.2       2023-12-06 [1] CRAN (R 4.3.2)
# lifecycle     1.0.4       2023-11-07 [1] CRAN (R 4.3.0)
# lubridate   * 1.9.3       2023-09-27 [1] CRAN (R 4.3.1)
# magrittr      2.0.3       2022-03-30 [1] CRAN (R 4.2.1)
# memoise       2.0.1       2021-11-26 [1] CRAN (R 4.2.1)
# mime          0.12        2021-09-28 [1] CRAN (R 4.2.0)
# miniUI        0.1.1.1     2018-05-18 [1] CRAN (R 4.2.1)
# munsell       0.5.0       2018-06-12 [1] CRAN (R 4.2.1)
# odbc        * 1.4.0.9000  2023-12-19 [1] Github (r-dbi/odbc@95e8b69)
# openxlsx    * 4.2.5.2     2023-02-06 [1] CRAN (R 4.2.2)
# pillar        1.9.0       2023-03-22 [1] CRAN (R 4.2.3)
# pkgbuild      1.4.3       2023-12-10 [1] CRAN (R 4.3.2)
# pkgconfig     2.0.3       2019-09-22 [1] CRAN (R 4.2.1)
# pkgload       1.3.3       2023-09-22 [1] CRAN (R 4.3.1)
# pool        * 1.0.1       2023-02-21 [1] CRAN (R 4.2.2)
# profvis       0.3.8       2023-05-02 [1] CRAN (R 4.3.0)
# promises      1.2.1       2023-08-10 [1] CRAN (R 4.3.1)
# purrr       * 1.0.2       2023-08-10 [1] CRAN (R 4.3.1)
# R6            2.5.1       2021-08-19 [1] CRAN (R 4.2.1)
# Rcpp          1.0.11      2023-07-06 [1] CRAN (R 4.3.1)
# readr       * 2.1.4       2023-02-10 [1] CRAN (R 4.2.2)
# remotes       2.4.2.1     2023-07-18 [1] CRAN (R 4.3.1)
# rlang       * 1.1.2       2023-11-04 [1] CRAN (R 4.3.2)
# rstudioapi    0.15.0      2023-07-07 [1] CRAN (R 4.3.1)
# scales        1.3.0       2023-11-28 [1] CRAN (R 4.3.2)
# sessioninfo   1.2.2       2021-12-06 [1] CRAN (R 4.2.1)
# shiny         1.8.0       2023-11-17 [1] CRAN (R 4.3.2)
# sodium      * 1.3.1       2023-11-20 [1] CRAN (R 4.3.2)
# stringi       1.8.3       2023-12-11 [1] CRAN (R 4.3.2)
# stringr     * 1.5.1       2023-11-14 [1] CRAN (R 4.3.2)
# tibble      * 3.2.1       2023-03-20 [1] CRAN (R 4.2.2)
# tidyr       * 1.3.0       2023-01-24 [1] CRAN (R 4.2.2)
# tidyselect    1.2.0       2022-10-10 [1] CRAN (R 4.2.1)
# tidyverse   * 2.0.0       2023-02-22 [1] CRAN (R 4.2.2)
# timechange    0.2.0       2023-01-11 [1] CRAN (R 4.2.2)
# tzdb          0.4.0       2023-05-12 [1] CRAN (R 4.3.0)
# urlchecker    1.0.1       2021-11-30 [1] CRAN (R 4.2.1)
# usethis       2.2.2       2023-07-06 [1] CRAN (R 4.3.1)
# utf8          1.2.4       2023-10-22 [1] CRAN (R 4.3.2)
# vctrs         0.6.5       2023-12-01 [1] CRAN (R 4.3.2)
# vroom         1.6.5       2023-12-05 [1] CRAN (R 4.3.2)
# withr         2.5.2       2023-10-30 [1] CRAN (R 4.3.2)
# xtable        1.8-4       2019-04-21 [1] CRAN (R 4.2.1)
# zip           2.3.0       2023-04-17 [1] CRAN (R 4.2.3)
# 
# [1] C:/Users/EC_E91367/Documents/R/lib
# [2] C:/Program Files/R/R-4.3.2/library
```

</details>

**Comment by ibecav** [comment]
Just discovered this issue is also true for Microsoft SQL server for the latest versions of odbc and DBI (not at all sure which is "at fault") but same symptoms no longer able to use the dbExistsTable() or other related functions.  Get back `unable to find an inherited method for function ‘dbExistsTable’ for signature ‘"Microsoft SQL Server", "NULL"’.  As I have time will send a better reprex but it's essentially the exact same issue only not Oracle but Microsoft.

**Comment by capitantyler** [comment]
@ibecav I give you this code to pass througth until the issue was solved:

```
box::use(
  rlang[arg_match, is_missing],
  glue[glue],
  dplyr[`%>%`, mutate_if, mutate, transmute],
  tidyr[unite],
  readr[format_delim],
  stringr[str_replace_all],
  DBI[dbGetQuery],
  purrr[map],
  lubridate[is.Date],
)
#' insert_into
#' 
#' add rows in destination_table by SQL INSERT INTO command
#'
#' @param conn            conectión
#' @param df_origin       df to insert into
#' @param destination_table   destination table name
#'
#' @return
#' @export
#'
#' @examples
insert_into <- function(
  conn,
  df_origin,
  destination_table
){

  df_insert <- df_origin %>% 
    mutate_if(
      is.Date,
      ~ glue("to_date('{format(., '%Y-%m-%d')}','YYYY-MM-DD')")
    )  %>% 
    tidyr::unite(
      "INSERT", everything(), sep = ", "
    ) %>% 
    transmute(
      INSERT = glue(
        "insert into {destination_table} ({
       paste0(
         names(df_origin), collapse = ','
       )}) values ({
       INSERT
       });"
      )
    )
  
  
  map(
    .x = df_insert$INSERT,
    .f = \ (x) {
      dbGetQuery(
        conn,
        statement = x
      )
    }
  )
  
}
```

**Comment by detule** [comment]
@capitantyler 

Do you have the ability to test the changes in [this PR](https://github.com/r-dbi/odbc/pull/712) to see if this resolves your ORACLE issues with `dbExistsTable`.  You need to use `DBI 1.1.3` ( not `1.2.0` ).

Thanks

**Comment by hadley** [comment]
@ibecav that seems unrelated to this issue. Can you please file a new issue with reprex? Thanks!

**Comment by ibecav** [comment]
@hadley will do. We rolled back to versions of the libraries that worked for us (previously and now) so please give me 24 hours so I can set things up to give you a proper reprex.

Side note-- thanks to you and all the others who keep the tidyverse going. Big fans here!

**Comment by capitantyler** [comment]
@detule , here you are, my reprex. Your [PR](https://github.com/r-dbi/odbc/pull/712) is working.

``` r
library(DBI)
library(odbc)  #detule

conn <- DBI::dbConnect(
  drv = odbc::odbc(), 
  dsn = mydsn,
  uid = myuid,
  PWD = mypwd 
)
#> Error in eval(expr, envir, enclos): objeto 'mydsn' no encontrado

dbIsValid(conn)
#> Error in h(simpleError(msg, call)): error in evaluating the argument 'dbObj' in selecting a method for function 'dbIsValid': objeto 'conn' no encontrado
DBI::dbGetInfo(conn)
#> Error in h(simpleError(msg, call)): error in evaluating the argument 'dbObj' in selecting a method for function 'dbGetInfo': objeto 'conn' no encontrado

# check existence VIA SQL
res <- DBI::dbGetQuery(
  conn,
   glue(
     "Select 1 from  sys.all_tables \
      where owner = '{myuid}' \
      and table_name = 'TIPO_CAMBIO'" 
   )
)
#> Error in h(simpleError(msg, call)): error in evaluating the argument 'conn' in selecting a method for function 'dbGetQuery': objeto 'conn' no encontrado
res
#> Error in eval(expr, envir, enclos): objeto 'res' no encontrado

# now is OK dbExistsTable(conn, "TIPO_CAMBIO") is TRUE
dbExistsTable(conn, "TIPO_CAMBIO")
#> Error in h(simpleError(msg, call)): error in evaluating the argument 'conn' in selecting a method for function 'dbExistsTable': objeto 'conn' no encontrado

dbExistsTable(conn, Id(
  schema = myuid, table = "TIPO_CAMBIO"
))
#> Error in h(simpleError(msg, call)): error in evaluating the argument 'conn' in selecting a method for function 'dbExistsTable': objeto 'conn' no encontrado

dbExistsTable(conn, Id(
  owner = myuid, table = "TIPO_CAMBIO"
))
#> Error in h(simpleError(msg, call)): error in evaluating the argument 'conn' in selecting a method for function 'dbExistsTable': objeto 'conn' no encontrado

# insert example
input <- tibble(
  CCY_FROM = "840",
  FECHA = ymd("2024-01-31"),
  TC = 20.25
)
#> Error in tibble(CCY_FROM = "840", FECHA = ymd("2024-01-31"), TC = 20.25): no se pudo encontrar la función "tibble"

dbWriteTable(
  conn = conn,
  name = "TIPO_CAMBIO",
  value = input,
  append = TRUE
)
#> Error in h(simpleError(msg, call)): error in evaluating the argument 'conn' in selecting a method for function 'dbWriteTable': objeto 'conn' no encontrado

DBI::dbGetQuery(
  conn,
  glue(
    "Select * from  TIPO_CAMBIO \
      where FECHA = DATE '2024-01-31'"
  )
)
#> Error in h(simpleError(msg, call)): error in evaluating the argument 'conn' in selecting a method for function 'dbGetQuery': objeto 'conn' no encontrado
# working OK

# These are my connection string and my TNSNAMES which I has been connecting succesfully:
#   
#   PRULES =
#   (DESCRIPTION =
#      (ADDRESS = (PROTOCOL = TCP)(HOST = exac2-scan1.lorenipsum.com.ar)(PORT = MyPort))
#    (CONNECT_DATA =
#        (SERVER = DEDICATED)
#      (SERVICE_NAME = myServiceName)
#    )
#   )
# 
# Driver={{Oracle en OraClient12Home1}};DBQ=exac2-scan1.lorenipsum.com.ar:MyPort/myServiceName;UID={usr};PWD={pwd};
sessionInfo()
#> R version 4.3.2 (2023-10-31 ucrt)
#> Platform: x86_64-w64-mingw32/x64 (64-bit)
#> Running under: Windows 10 x64 (build 19045)
#> 
#> Matrix products: default
#> 
#> 
#> locale:
#> [1] LC_COLLATE=Spanish_Argentina.utf8  LC_CTYPE=Spanish_Argentina.utf8   
#> [3] LC_MONETARY=Spanish_Argentina.utf8 LC_NUMERIC=C                      
#> [5] LC_TIME=Spanish_Argentina.utf8    
#> 
#> time zone: America/Buenos_Aires
#> tzcode source: internal
#> 
#> attached base packages:
#> [1] stats     graphics  grDevices utils     datasets  methods   base     
#> 
#> other attached packages:
#> [1] odbc_1.3.1.9000 DBI_1.2.0      
#> 
#> loaded via a namespace (and not attached):
#>  [1] vctrs_0.6.5       cli_3.6.2         knitr_1.45        rlang_1.1.2      
#>  [5] xfun_0.41         purrr_1.0.2       styler_1.10.2     glue_1.6.2       
#>  [9] bit_4.0.5         htmltools_0.5.7   hms_1.1.3         rmarkdown_2.25   
#> [13] R.cache_0.16.0    evaluate_0.23     fastmap_1.1.1     yaml_2.3.8       
#> [17] lifecycle_1.0.4   compiler_4.3.2    blob_1.2.4        fs_1.6.3         
#> [21] pkgconfig_2.0.3   Rcpp_1.0.11       rstudioapi_0.15.0 R.oo_1.25.0      
#> [25] R.utils_2.12.3    digest_0.6.33     reprex_2.0.2      magrittr_2.0.3   
#> [29] R.methodsS3_1.8.2 tools_4.3.2       withr_2.5.2       bit64_4.0.5
```

<sup>Created on 2024-01-04 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by hadley** [comment]
Fixed by #712.
